### PR TITLE
pyopencl: add new package

### PR DESCRIPTION
**Pull request by matthiasdiener**

### CODE REVIEW

**Comment by matthiasdiener** [comment]
CC: @inducer

File: var/spack/repos/builtin/packages/py-pyopencl/package.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+
+from spack import *
+
+
+class PyPyopencl(PythonPackage):
+    """Python wrapper for OpenCL."""
+
+    homepage = "https://documen.tician.de/pyopencl/"
+    url      = "https://files.pythonhosted.org/packages/75/ee/b8c71784fe0eb6997b5daf6065136ea7a8e64118a079917b0eeb70ed0d00/pyopencl-2020.2.2.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyopencl/package.py]: https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#url

File: var/spack/repos/builtin/packages/py-pyopencl/package.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+    maintainers = ['matthiasdiener']
+
+    version('2020.2.2', sha256='31fcc79fb6862998e98d91a624c0bd4f0ab4c5d418d199912d4d312c64e437ec')
+
+    depends_on('opencl')
+    depends_on('ocl-icd')
+    depends_on('py-mako')
+    depends_on('py-pybind11')
+    depends_on('py-numpy')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyopencl/package.py]: Missing build dep on setuptools, missing build/run dep on pytools and decorator

File: var/spack/repos/builtin/packages/py-pyopencl/package.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    homepage = "https://documen.tician.de/pyopencl/"
+    url      = "https://files.pythonhosted.org/packages/75/ee/b8c71784fe0eb6997b5daf6065136ea7a8e64118a079917b0eeb70ed0d00/pyopencl-2020.2.2.tar.gz"
+
+    maintainers = ['matthiasdiener']
+
+    version('2020.2.2', sha256='31fcc79fb6862998e98d91a624c0bd4f0ab4c5d418d199912d4d312c64e437ec')
+
+    depends_on('opencl')
+    depends_on('ocl-icd')
+    depends_on('py-mako')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyopencl/package.py]:
```suggestion
    depends_on('py-mako@0.3.6:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-pyopencl/package.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    depends_on('py-pybind11')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyopencl/package.py]:
```suggestion
    depends_on('py-pybind11@2.5.0:', type='build')
```

File: var/spack/repos/builtin/packages/py-pyopencl/package.py (hunk 2 above)
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyopencl/package.py]:
```suggestion
    depends_on('py-numpy', type=('build', 'run'))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-pyopencl/package.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+    homepage = "https://documen.tician.de/pyopencl/"
+    url      = "https://pypi.io/packages/source/p/pyopencl/pyopencl-2020.2.2.tar.gz"
+
+    maintainers = ['matthiasdiener']
+
+    version('2020.2.2', sha256='31fcc79fb6862998e98d91a624c0bd4f0ab4c5d418d199912d4d312c64e437ec')
+
+    depends_on('opencl')
+    depends_on('ocl-icd')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyopencl/package.py]: Missing dep:
```python
depends_on('python@3.6:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-pyopencl/package.py [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    depends_on('py-mako@0.3.6:', type=('build', 'run'))
+    depends_on('py-pybind11@2.5.0:', type='build')
+    depends_on('py-numpy', type=('build', 'run'))
+    depends_on('py-setuptools', type='build')
+    depends_on('py-pytools', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyopencl/package.py]:
```suggestion
    depends_on('py-pytools@2017.6:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-pyopencl/package.py [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    depends_on('py-decorator', type=('build', 'run'))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyopencl/package.py]:
```suggestion
    depends_on('py-decorator@3.2.0:', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-pyopencl/package.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyopencl/package.py:30]: Missing dep:
```python
depends_on('py-appdirs@1.4.0:', type=('build', 'run'))
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-pyopencl/package.py (hunk 8 above)
Comment (matthiasdiener) [reply var/spack/repos/builtin/packages/py-pyopencl/package.py:30]: Added.

File: var/spack/repos/builtin/packages/py-pyopencl/package.py (hunk 2 above)
Comment (matthiasdiener) [reply var/spack/repos/builtin/packages/py-pyopencl/package.py]: Added

**Review by adamjstewart — APPROVED** [review]

**Review by alalazo — APPROVED** [review]

**Pull request merged**